sdk/core/Azure.Core/src/Shared/OperationInternal.cs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class OperationInternal<TResult>
```

Comment:
Naming it `OperationInternal` for now. We can't use `OperationHelpers` because we still have some clients using the name.

Reply:
We currently have: `OperationInternal<TResult>` and `OperationInternal<TResult, TResponseType>`.
- `OperationInternal<TResult>` receives a `Response` from the service, and exposes the `Value` property as a `TResult`.
- `OperationInternal<TResult, TResponseType>` receives a `Response<ResponseType>` from the service, and exposes the `Value` property as a `TResult`.

Reply:
Yep. That's the plan. We'll probably keep the `OperationHelpers` name as well.